src/ConnectedMode/BranchMatcher.cs

```diff
@@ -40,21 +41,21 @@ public interface IBranchMatcher
         /// This is the same algorithm as the other SonarLint flavours (?except in how we treated branch histories.
         /// We treat branch histories as series of linear commits ordered by time, even if there are merges).
         /// </remarks>
-        Task<string> GetMatchedBranch(string projectKey);
+        Task<string> GetMatchingBranch(string projectKey, IRepository gitRepo);
```

Comment:
Two changes to make the class easier to use from the `ServerBranchMapper`:
1) MEF exportable so it can be created easily
2) passing the repo as an argument so that we don't need to create a new instance every time the git branch changes.